library/std/src/process.rs

```diff
@@ -2019,6 +2019,36 @@ impl Child {
         self.handle.kill()
     }
 
+    /// Ask a child process to exit.
+    ///
+    /// On Unix platforms this sends SIGTERM, which can be blocked, handled, or ignored.
+    ///
+    /// On Windows, and other platforms where there isn't a corrollary to SIGTERM, this just
+    /// executes `Child::kill`.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```no_run
+    /// #![feature(process_terminate)]
+    /// use std::process::Command;
+    ///
+    /// let mut command = Command::new("yes");
+    /// if let Ok(mut child) = command.spawn() {
+    ///     if let Err(e) = child.terminate() {
+    ///         println!("Child didn't respond to terminate");
+    ///         child.kill().expect("command couldn't be killed");
+    ///     }
+    /// } else {
+    ///     println!("yes command didn't start");
```

Comment:
Also, this example I wrote isn't very good....I meant to write something where the process doesn't exit after terminate, so then we escalate to kill, but there's no `.wait()` call in here, and `terminate()` errors would be from sending the signal, not waiting on the process to exit.